src/backend/app/auth/roles.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2022, 2023 Humanitarian OpenStreetMap Team
+#
+# This file is part of FMTM.
+#
+#     FMTM is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     FMTM is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with FMTM.  If not, see <https:#www.gnu.org/licenses/>.
+#
+
+"""User roles authorisation Depends methods.
+
+These methods use FastAPI Depends for dependency injection
+and always return an AuthUser object in a standard format.
+"""
+
+from fastapi import Depends, HTTPException
+from loguru import logger as log
+from sqlalchemy.orm import Session
+
+from app.auth.osm import AuthUser, login_required
+from app.db.database import get_db
+from app.db.db_models import DbProject, DbUser, DbUserRoles
+from app.models.enums import HTTPStatus, ProjectRole, UserRole
+from app.projects.project_deps import get_project_by_id
+
+
+async def get_uid(user_data: AuthUser) -> int:
+    """Extract user id from returned OSM user."""
+    if user_id := user_data.get("id"):
+        return user_id
+    else:
+        log.error(f"Failed to get user id from auth object: {user_data}")
+        raise HTTPException(
+            status_code=HTTPStatus.UNAUTHORIZED,
+            detail="Auth failed. No user id present",
+        )
+
+
+async def super_admin(
+    db: Session = Depends(get_db),
+    user_data: AuthUser = Depends(login_required),
+) -> AuthUser:
+    """Super admin role, with access to all endpoints."""
+    user_id = await get_uid(user_data)
+
+    match = db.query(DbUser).filter_by(id=user_id, role=UserRole.ADMIN).first()
+
+    if not match:
+        log.error(f"User ID {user_id} requested an admin endpoint, but is not admin")
+        raise HTTPException(
+            status_code=HTTPStatus.FORBIDDEN, detail="User must be an administrator"
+        )
+
+    return user_data
```

Comment:
Hi @spwoodcock 
What is the purpose of this function? 
Does this check if the user is super_admin or not? If that is the case, we might need to do something like 
`user.role == UserRole.ADMIN` after we get the user from db.
How does this work?

Reply:
Doesn't this line do that? https://github.com/hotosm/fmtm/blob/35eaa0d1b9fab5ee769f9e751384f348a8c59421/src/backend/app/auth/roles.py#L55

I definitely could have got something wrong here though - thanks for the sanity check!

Reply:
@spwoodcock my understanding is that this line filters data from user role with id=user_id and role=admin.
But, we need to do this.
```
user_role = db.query(DbUser).filter_by(id=user_id,).first().role

if user_role == UserRole.ADMIN:
    return True
else:
    return False
```

Reply:
They are basically the same thing.

The line further down does a check `if match`.
- If no user with the user id and role `admin` exists, then return a HTTPException.
- If a user exists and has the role `admin`, then continue to return the user details.

Does that make sense?